### ISSUE TITLE
Dynamic masking imageset.get_mask() broken

### ISSUE THREAD

**Issue by rjgildea** [dials/dials#841]
test/util/test_masking.py::test_dynamic_shadowing is currently broken after the dynamic masking refactoring.

The masker itself is working as expected:
```
masker = imageset.masker()
m1 = masker.get_mask(detector, scan.get_oscillation()[0])
pyplot.imshow(m1[0].as_numpy_array())
```
<img width="438" alt="Screenshot 2019-07-05 at 11 23 31" src="https://user-images.githubusercontent.com/2810624/60716738-0e551800-9f18-11e9-89ea-0b75b2b093e3.png">

However calling `imageset.get_mask()` no longer works as before, i.e. it now only masks the dead pixels/gaps, and doesn't include the dynamic mask:
```
m2 = imageset.get_mask(0)
pyplot.imshow(m2[0].as_numpy_array())
```
<img width="432" alt="Screenshot 2019-07-05 at 11 23 57" src="https://user-images.githubusercontent.com/2810624/60716730-0b5a2780-9f18-11e9-990a-6114ccf15682.png">

Dynamic masks are no longer shown in the image_viewer with `show_mask=True`:
```
$ dials.import $(libtbx.find_in_repositories dials_regression)/shadow_test_data/DLS_I04_SmarGon/Th_3_O45_C45_P48_1_0500.cbf dynamic_shadowing=True
$ dials.image_viewer imported_experiments.json show_mask=True
```
<img width="633" alt="Screenshot 2019-07-05 at 11 31 12" src="https://user-images.githubusercontent.com/2810624/60716911-6ab83780-9f18-11e9-8d4b-2f81583b352d.png">

**Comment by Anthchirp** [comment]
Replacing
https://github.com/cctbx/dxtbx/blob/ef5eae1a80927f53cd6ae2fa1bca42a1f327a044/imageset.h#L1259
with
```C++
double scan_angle = 94.9;
```
where
```python
>>> scan.get_oscillation()[0]
94.90000000000002
```
fixes the issue.

**Comment by Anthchirp** [comment]
Two issues here.

First issue: `masker->get_mask` expected degrees, whereas in C++-land the common standard to pass values around appears to be in radians. Naive fix for this incoming. Proper fix should probably move the C++ masker interface to expect radians.

Second issue: scan object appears to be inconsistent. The image number 500 out of the scan starts at an oscillation angle of 94.9° as per image headers. 
```
>>> scan.get_oscillation()
(94.90000000000002, 0.1)  # correct
>>> scan.get_image_range()
(500, 500)  # correct
>>> scan.get_angle_from_image_index(500)
94.90000000000002  # correct
>>> scan.get_angle_from_array_index(0)
45.000000000000014  # problem
```
`get_dynamic_mask()` uses `get_angle_from_array_index()`.
`get_angle_from_array_index(n)` returns `get_angle_from_image_index(n+1)`.
Should this be `get_angle_from_image_index(n+image_range[0])`?

**Comment by jmp1985** [comment]
@Anthchirp Regarding scan array range see issue https://github.com/dials/dials/issues/277. I think that array_range, image_range is kind of confusing anyway so maybe array range should be removed.

This has always been confusing and tbh I suspect some things would break if you modified get_angle_from_image_index like that.

**Comment by jmp1985** [comment]
$ grep -r "array_range" . --include="*.py" --include="*.h" --include="*.cc" | wc -l
127

Looks like there are only 127 references to array_range in dials so fixing all these places will take some time but not an inordinate amount of time.

**Comment by graeme-winter** [comment]
Suspect this could affect things like cctbx.xfel as well - may be worth doing a deeper grep...

**Comment by jmp1985** [comment]
I think that the scan thing shouldn't affect xfel surely?

**Comment by graeme-winter** [comment]
Array range might…

**Comment by jmp1985** [comment]
I have now fixed this issue in dxtbx c7be9fb1ae1fb817248f33749d601249b62f9ae3

**Comment by phyy-nx** [comment]
Dropping array range from scans wouldn't affect stills processing.  It
could affect some side projects but those are readily adapted.
